librepo/downloader.c

```diff
@@ -704,23 +711,22 @@ select_suitable_mirror(LrDownload *dd,
     *selected_mirror = NULL;
     // mirrors_iterated is used to allow to use mirrors multiple times for a target
     unsigned mirrors_iterated = 0;
+    // retry local paths have no reason
+    gboolean reiterate = FALSE;
     //  Iterate over mirrors for the target. If no suitable mirror is found on
     //  the first iteration, relax the conditions (by allowing previously
     //  failing mirrors to be used again) and do additional iterations up to
     //  number of allowed failures equal to dd->allowed_mirror_failures.
     do {
-        // Sleep when all mirrors were already tried
-        // It prevents to call same url immediately after fail
-        if (mirrors_iterated) {
-            sleep(mirrors_iterated);
-        }
         // Iterate over mirror for the target
         for (GSList *elem = target->lrmirrors; elem; elem = g_slist_next(elem)) {
             LrMirror *c_mirror = elem->data;
             gchar *mirrorurl = c_mirror->mirror->url;
 
             // first iteration, filter out mirrors that failed previously
             if (mirrors_iterated == 0) {
+                if (c_mirror->mirror->protocol != LR_PROTOCOL_FILE)
```

Comment:
So, if one of mirror is not local then reiteration is enabled. But then local paths are retried too, isn't it? You wrote "retry local paths nave no reason" above. I think you forgot add condition code at line 743. 
```
            } else if (c_mirror->mirror->protocol == LR_PROTOCOL_FILE) {
                continue;
```
Or am I misunderstand the code?